### ISSUE TITLE
Make cutflow MC bare work again

### ISSUE THREAD

**Issue by yipengsun** [umd-lhcb/lhcb-ntuples-gen#73]
Our latest production of cutflow bare doesn't seem to have right numbers. I'll investigate them further to re-produce a final version of the cutflow MC bare ntuples.

I also want to use this opportunity to remove all obsolete flags.

## Run 1
- The Muon-TIS filtering was turned on for run 1, resulting in a ~97% efficiency for L0 trigger

## Run 2
- Additional HLT2 cut always applied

**Comment by yipengsun** [comment]
For run 1, after the bugfix, the cutflow looks like this:
```yml
Total events:
    input: 17894
    output: 17894
Relaxed $D^0 \mu$ combo:
    input: 3920
    output: 2417
$D^{*+} \rightarrow D^0 \pi^+$:
    input: 2417
    output: 2242
$\bar{B}^0 \rightarrow D^{*+} \mu^-$:
    input: 2109
    output: 2045
L0:
    input: 2045
    output: 806
Hlt1:
    input: 806
    output: 400
Hlt2:
    input: 400
    output: 163
Stripping:
    input: 163
    output: 80
DaVinci $D^* \mu$ cuts:
    input: 80
    output: 57
Offline $D^0$ cuts:
    input: 57
    output: 32
Offline $\mu$ cuts:
    input: 32
    output: 25
Offline $D^* \mu$ combo cuts:
    input: 25
    output: 25
```

**Comment by yipengsun** [comment]
For updated run 2 bare, here's the cutflow (the generation of the cutflow ntuples is now very slow, consistent w/ my old impression). Now all trigger cuts have lower efficiency compared to run 1:
```yml
Total events:
    input: 26633
    output: 26633
Relaxed $D^0 \mu$ combo:
    input: 22819
    output: 9154
$D^{*+} \rightarrow D^0 \pi^+$:
    input: 9154
    output: 8699
$\bar{B}^0 \rightarrow D^{*+} \mu^-$:
    input: 8351
    output: 8119
L0:
    input: 8119
    output: 2809
Hlt1:
    input: 2809
    output: 1310
Hlt2:
    input: 1310
    output: 421
Stripping:
    input: 421
    output: 339
DaVinci $D^* \mu$ cuts:
    input: 339
    output: 196
Offline $D^0$ cuts:
    input: 196
    output: 58
Offline $\mu$ cuts:
    input: 58
    output: 49
Offline $D^* \mu$ combo cuts:
    input: 49
    output: 43
```

**Comment by yipengsun** [comment]
One thing I noticed is that the bare cuts between run 1 and run 2 are not fully sync'ed yet. So this is not an apple-to-apple comparison. I'll try sync them (use the looser of the two), and try again.

**Comment by yipengsun** [comment]
With updated bare ntuples, we have:

| cut name                             | run 1 yield   | run 2 yield   | run 1 efficiency   | run 2 efficiency   | double ratio   |
|--------------------------------------|---------------|---------------|--------------------|--------------------|----------------|
| Total events                         | 17894         | 26633         | -                  | -                  | -              |
| Relaxed $D^0 \mu$ combo              | 2417          | 3736          | 61.7               | 61.6               | 1.00           |
| $D^{*+} \rightarrow D^0 \pi^+$       | 2242          | 3542          | 92.8               | 94.8               | 1.02           |
| $\bar{B}^0 \rightarrow D^{*+} \mu^-$ | 2045          | 3295          | 97.0               | 97.6               | 1.01           |
| L0                                   | 806           | 1052          | 39.4               | 31.9               | 0.81           |
| Hlt1                                 | 400           | 753           | 49.6               | 71.6               | 1.44           |
| Hlt2                                 | 163           | 379           | 40.8               | 50.3               | 1.24           |
| Stripping                            | 80            | 324           | 49.1               | 85.5               | 1.74           |
| DaVinci $D^* \mu$ cuts               | 57            | 188           | 71.2               | 58.0               | 0.81           |
| Offline $D^0$ cuts                   | 32            | 56            | 56.1               | 29.8               | 0.53           |
| Offline $\mu$ cuts                   | 25            | 47            | 78.1               | 83.9               | 1.07           |
| Offline $D^* \mu$ combo cuts         | 25            | 41            | 100.0              | 87.2               | 0.87           |
| Total ratio                          | -             | -             | 0.1                | 0.2                | 1.10           |

A few observations:
- The very loose BARE cuts in `DaVinci` has very similar efficiencies for both run 1 and 2
- The main efficiency loss for run 2 comes from offline D0 cuts
- The trigger efficiencies now have much better agreement w/ our expectations.

**Comment by manuelfs** [comment]
Thank you very much for doing the cutflow, and also included useful observations. 👍

Which ntuples is this cutflow done with, the full 1M cocktail? Any truthmatching? The numbers are very different from any table done with the old cocktail ntuples.

**Comment by yipengsun** [comment]
These are locally generated small bare ntuples. No truth-matching is performed.

**Comment by yipengsun** [comment]
I'm still validating MC and real data reconstruction to make sure everything still works. Do you want me to re-submit cocktail bare again if these validations are successful?

BTW, I'm glad that I did a side-by-side editing for our run 1 and run 2 DaVinci scripts: I fixed lots of formatting inconsistencies between the two and now they should be maximally similar.

**Comment by yipengsun** [comment]
Both run 1 and run 2 DaVinci script validated. Will re-submit bare.

BTW, I plan to remove the faulty bare ntuples that are generated recently (the bare ntuples in the `0.9.4` folder). Given that they are useless, deleting them can save some disk space.

**Comment by yipengsun** [comment]
@manuelfs Here's the latest cutflow bare tables: https://umd-lhcb.github.io/lhcb-ntuples-gen/data/cutflows/RDst_run1_vs_run2/overview/

They have much better agreement w/ our expectations.

**Comment by yipengsun** [comment]
`11874094` doesn't seem to be produced:
```
❯ ./size_mc_samples.py -i 11874094 -m detail
Before proceed, don't forget to run lhcb-proxy-init!!
For MC ID 11874094

```

The result is empty.

**Comment by manuelfs** [comment]
> ## Run 2
> 
>     * Additional HLT2 cut always applied

@yipengsun What do you mean here, that the old cutflow BARE ntuples had some HLT2 cut always applied? If so, which one was it?

**Comment by manuelfs** [comment]
Ah, I think he meant not one but all the HLT cuts that he removed in `BARE` 0.9.4, documented here
![image](https://user-images.githubusercontent.com/4977423/126326379-66d0dced-bd45-4f2f-82d8-6eb3f9aea986.png)

**Comment by manuelfs** [comment]
And that fully explains the discrepancy in the HLT2 efficiency between the 0.9.0 and 0.9.4 ntuples. To recap, with the 0.9.0 ntuples that we used in the May 26 talk to the Semileptonic group, we had a 54% increase in HLT2 efficiency for signal
![image](https://user-images.githubusercontent.com/4977423/126329059-900eb40f-643c-4367-b6b9-a9a0851ceda9.png)

However, with the 0.9.4 ntuples that increase was gone, even though we were cutting on the same `b0_Hlt2XcMuXForTauB2XcMuDecision_TOS` variable
![image](https://user-images.githubusercontent.com/4977423/126329242-b664466b-83be-45a8-90f0-b95344f321f1.png)

This is because, as the table in the previous post shows, the 0.9.4 BARE cuts removed the PID that the 0.9.0 ntuples had. As a result, HLT2 cuts more events, but theses are events that were going to be cut by the stripping anyway (which is why stripping goes from 51% to 80% more efficient). 

In other words, the new HLT2 is not more efficient than the old trigger despite the being targeted to a B decay with the looser DIRA, but this is because it already includes some PID cuts that were going to be applied in the stripping anyway
![image](https://user-images.githubusercontent.com/4977423/126329870-dc725e5b-3325-4d7d-922c-3eed16efeceb.png)

I checked this by applying the HLT2 cut in MD `BARE` ntuples. In 0.9.0, we have 1963 signal events, and 405 pass HLT2 (20.6%). In 0.9.4, 371/2310 = 16.1%.  However, if I apply the `k_PIDK > 2 & pi_PIDK < 4 & mu_isMuon & mu_PIDmu > -400` cuts first, in 0.9.0 we still have 405/1963 = 20.6% but in 0.9.4 we have 367/1763 = 20.8%. I'm not sure why 0.9.4 has fewer events, but it is not a big deal.
